library/VTKExtensions/Rendering/vtkF3DPolyDataMapper.cxx

```diff
@@ -180,3 +180,53 @@ void vtkF3DPolyDataMapper::ReplaceShaderValues(
 
   this->Superclass::ReplaceShaderValues(shaders, ren, actor);
 }
+
+//-----------------------------------------------------------------------------
+bool vtkF3DPolyDataMapper::RenderWithMatCap(vtkActor* actor)
+{
+  // check if we have normals
+  if (this->VBOs->GetNumberOfComponents("normalMC") != 3)
+  {
+    return false;
+  }
+
+  auto textures = actor->GetProperty()->GetAllTextures();
+
+  auto fn = [](const std::pair<std::string, vtkTexture*>& tex) { return tex.first == "matcap"; };
+
+  return std::find_if(textures.begin(), textures.end(), fn) != textures.end();
```

Comment:
This is a O(n) call, can we avoid calling that at every ReplaceShaderColor/ReplaceShaderLight call ?

Reply:
It's called once per actor when the shader is built and I've never seen a use case with more than 5 textures, I think we're good here.